kernel/src/main/java/org/kframework/kompile/DefinitionParsing.java

```diff
@@ -355,74 +346,150 @@ private Module resolveConfigBubbles(Definition def, Module inputModule, RuleGram
+        Map<Module, ParseInModule> donorScanners = new HashSet<>(donorModule.values()).parallelStream().map(x -> {
+            ParseInModule pim = RuleGrammarGenerator.getCombinedGrammar(gen.getRuleGrammar(x), isStrict, profileRules, files);
+            pim.getScanner(options.global);
+            return new Tuple2<>(x, pim);
+        }).collect(Collectors.toMap(Tuple2::_1, Tuple2::_2));
+
+        // do parsing on remaining bubbles and collect everything in `parsedSentences`
+        // this way we can easily parallelize the steps and we don't have to deal with the complex structure of modules
+        Map<String, java.util.Set<Sentence>> parsedSentences = stream(defWithCaches.modules())
+                .parallel()
+                .flatMap(m -> {
+                    if (stream(m.localSentences()).noneMatch(s -> s instanceof Bubble))
+                        return Stream.of();
+                    try (ParseInModule pim = RuleGrammarGenerator.getCombinedGrammar(gen.getRuleGrammar(m), isStrict, profileRules, files)) {
+                        pim.setScanner(donorScanners.get(donorModule.get(m.name())).getScanner());
+                        pim.initialize();
+                        ParseCache cache = loadCache(pim.seedModule());
+                        java.util.Set<Sentence> sentences = stream(m.localSentences())
+                                .filter(s -> s instanceof Bubble)
+                                .map(s -> (Bubble) s)
+                                .parallel()
+                                .flatMap(b -> {
+                                    Tuple2<Either<java.util.Set<KEMException>, K>, java.util.Set<KEMException>> result;
+                                    int startLine = b.att().get("contentStartLine", Integer.class);
+                                    int startColumn = b.att().get("contentStartColumn", Integer.class);
+                                    Source source = b.att().get(Source.class);
+                                    result = pim.parseString(b.contents(), START_SYMBOL, pim.getScanner(), source, startLine, startColumn, true, b.att().contains(Att.ANYWHERE()) || b.att().contains(Att.SIMPLIFICATION()) || ExpandMacros.isMacro(b));
+                                    parsedBubbles.getAndIncrement();
+                                    if (kem.options.warnings2errors && !result._2().isEmpty()) {
+                                        for (KEMException err : result._2()) {
+                                            if (kem.options.includesExceptionType(err.exception.getType())) {
+                                                errors.add(KEMException.asError(err));
+                                            }
+                                        }
+                                    } else {
+                                        kem.addAllKException(result._2().stream().map(e -> e.getKException()).collect(Collectors.toList()));
+                                    }
```

Comment:
This same logic is used here and above to deal with both `warnings2Errors` and warnings themselves. Can that be factored out to a method to make this shorter? (line 384 above)